code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -269,11 +275,14 @@
 	X.mouse_opacity = initial(X.mouse_opacity)
 	X.density = TRUE
 	X.allow_pass_flags &= ~PASSABLE
-	REMOVE_TRAIT(X, TRAIT_IMMOBILE, WIDOW_ABILITY_TRAIT)
-	REMOVE_TRAIT(X, TRAIT_BURROWED, WIDOW_ABILITY_TRAIT)
-	REMOVE_TRAIT(X, TRAIT_HANDS_BLOCKED, WIDOW_ABILITY_TRAIT)
+	REMOVE_TRAIT(X, TRAIT_IMMOBILE, ability_name)
+	REMOVE_TRAIT(X, TRAIT_BURROWED, ability_name)
+	REMOVE_TRAIT(X, TRAIT_HANDS_BLOCKED, ability_name)
 	X.update_icons()
 	add_cooldown()
+	if(!owner.buckled_mobs)
+		return
```

Comment:
Unnecesarry check.

Reply:
The check exists because other code handles it the same way, but I've removed it